httprule/client.go

```diff
@@ -0,0 +1,118 @@
+package httprule
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	pb "google.golang.org/genproto/googleapis/api/annotations"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	descpb "google.golang.org/protobuf/types/descriptorpb"
+)
+
+type ClientConn struct {
+	HTTPClient *http.Client
+	BaseURL    string
+}
+
+var (
+	ErrInvalidMethod    = errors.New("invalid gRPC method string")
+	ErrMethodNotFound   = errors.New("method not found")
+	ErrServiceNotFound  = errors.New("service not found")
+	ErrNotImplemented   = errors.New("not implemented")
+	ErrHttpRuleNotFound = errors.New("no HttpRule")
+)
+
+func (c *ClientConn) NewStream(ctx context.Context, desc *grpc.StreamDesc, method string, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+	return nil, ErrNotImplemented
+}
+
+func (c *ClientConn) Invoke(ctx context.Context, method string, args, reply interface{}, opts ...grpc.CallOption) error {
+	rule, err := getHttpRule(method)
+	if err != nil {
+		return err
+	}
+	req, err := NewHTTPRequest(rule, c.BaseURL, args.(proto.Message))
+	if err != nil {
+		return err
+	}
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
```

Comment:
Does this need a `nil` check?

Reply:
Nope (I know I said that to you and then I checked).
I thought it did, but that was only on the request e.g. server side....
See https://pkg.go.dev/net/http#Response

 	// The http Client and Transport guarantee that Body is always
	// non-nil, even on responses without a body or responses with
	// a zero-length body. It is the caller's responsibility to
	// close Body. The default HTTP client's Transport may not
	// reuse HTTP/1.x "keep-alive" TCP connections if the Body is
	// not read to completion and closed.